### ISSUE TITLE
Type data lost when `customInstance` returns more than just the model's response

### ISSUE THREAD

**Issue by alb7h** [anymaniax/orval#199]
## What are the steps to reproduce this issue?
1. Have custom instance return anything other than the exact resource

## What happens?

Hooks lose type data

## What were you expecting to happen?

Hooks would contain the new Promise return type of the customInstance

## Any logs, error output, etc?

No

## Any other comments?

This is my customInstance fn:

```ts
import Axios, { AxiosRequestConfig } from 'axios';

export const AXIOS_INSTANCE = Axios.create({
  baseURL: process.env.NEXT_PUBLIC_API_URL,
});

export const customInstance = <TData>(
  config: AxiosRequestConfig
): Promise<{ data: TData; headers: Record<string, string> }> => {
  const source = Axios.CancelToken.source();
  const promise = AXIOS_INSTANCE.request<TData>({
    ...config,
    cancelToken: source.token,
  }).then(({ data, headers }) => ({
    data,
    headers,
  }));

  (promise as any).cancel = () => {
    source.cancel('Query was cancelled by React Query');
  };

  return promise;
};

export default customInstance;
```

When I try to return more than just the exact data, all hooks lose type information in their consumers. I believe this is because the query functions generated don't actually assert `TData` is the model it wants, it just states that TData defaults to the model it wants. 

Using [this](https://github.com/anymaniax/orval/blob/master/samples/react-app-with-react-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts#L37) as an example, I think the generic should be `TData extends Pets = Pets` which validates and defaults to Pets vs `TData = Pets` which can be changed to anything at all.

Making this change fixes things for me, but I'm not 100% that it's the correct solution given what `AsyncReturnType` is trying to do.

## What versions are you using?
**Operating System:** …
**Package Version:** …
**Browser Version:** …

**Comment by anymaniax** [comment]
Hello @alb7h, thanks for the issue and I am currently checking the problem. I will probably solve this in the next major version that I am currently working on. I currently solve it like [this](https://gist.github.com/anymaniax/6d5d9f633b546212cc3db8981af0732e) but doing this limit us so I am currently trying to find a better solution to this one

**Comment by anymaniax** [comment]
Fixed with the `6.0.0`. You can try it now with `orval@next`

**Comment by anymaniax** [comment]
I close the issue leave me a comment if you have any problem
